scripts/abins/input/abinitioloader.py

```diff
@@ -138,14 +142,14 @@ def load_formatted_data(self):
         return self._rearrange_data(data=loaded_data)
 
     # Internal method for use by child classes which read ab initio phonon data
-    def _rearrange_data(self, data=None):
+    @staticmethod
+    def _rearrange_data(data: dict) -> abins.AbinsData:
```

Comment:
is it ok that we have removed the default value for data?

Reply:
I think so; it's useless to call this function with `data=None` and will immediately lead to a KeyError.

A lot of functions in Abins were written with non-functioning `None` default parameters; I think the intent behind this was to implement mandatory keyword arguments. Python has a proper syntax for that since Python 3 (https://peps.python.org/pep-3102/) but at the time Abins still supported Python 2.7.

I think the intent behind using a pseudo-required-kwarg was to improve readability by using explicitly named arguments in function calls. But I don't agree that adding the word "data" is beneficial for a function like this one that only takes one argument, so while adding the type hints it seems a good chance to simplify things as well.

The positional argument will still be backward-compatible with any code still using `data=...`; it is not a [positional-only parameter](https://peps.python.org/pep-0570/).